src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -249,29 +329,438 @@ public DateTime(int year, int month, int day, int hour, int minute, int second,
+        /// <param name="microsecond">The milliseconds (0 through 900).</param>
+        /// <param name="calendar">The calendar that is used to interpret <paramref name="year"/>, <paramref name="month"/>, and <paramref name="day"/>.</param>
+        /// <param name="kind">
+        /// One of the enumeration values that indicates whether <paramref name="year"/>, <paramref name="month"/>, <paramref name="day"/>,
+        /// <paramref name="hour"/>, <paramref name="minute"/>, <paramref name="second"/>, and <paramref name="millisecond"/>
+        /// specify a local time, Coordinated Universal Time (UTC), or neither.</param>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="calendar"/> is <see langword="null"/>
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// <paramref name="year"/> is not in the range supported by <paramref name="calendar"/>.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="month"/> is less than 1 or greater than the number of months in <paramref name="calendar"/>.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="day"/> is less than 1 or greater than the number of days in <paramref name="month"/>.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="hour"/> is less than 0 or greater than 23.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="minute"/> is less than 0 or greater than 59.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="second"/> is less than 0 or greater than 59.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="millisecond"/> is less than 0 or greater than 999.
+        ///
+        /// -or-
+        ///
+        /// <paramref name="microsecond"/> is less than 0 or greater than 900.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="kind"/> is not one of the <see cref="DateTimeKind"/> values.
+        /// </exception>
+        /// <remarks>
+        /// The allowable values for <paramref name="year"/>, <paramref name="month"/>, and <paramref name="day"/> parameters
+        /// depend on the <paramref name="calendar"/> parameter. An exception is thrown if the specified date and time cannot
+        /// be expressed using <paramref name="calendar"/>.
+        ///
+        /// For applications in which portability of date and time data or a limited degree of time zone awareness is important,
+        /// you can use the corresponding <see cref="DateTimeOffset"/> constructor.
+        /// </remarks>
+        public DateTime(int year, int month, int day, int hour, int minute, int second, int millisecond, int microsecond, Calendar calendar!!, DateTimeKind kind)
+            : this(year, month, day, hour, minute, second, millisecond, calendar, kind)
+        {
+            if (microsecond is < 0 or >= MicrosecondsPerMillisecond)
```

Comment:
>if (microsecond is < 0 or >= MicrosecondsPerMillisecond) [](http://example.com/codeflow?start=12&length=56)

nit: maybe we can write this as:
```C#
 if ((uint)microsecond >= MicrosecondsPerMillisecond)
```

Reply:
But this would permit passing a negative `microsecond`.

```cs
var a = -1;
var b = (uint)a; // = 4294967295
```

<https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEUCuA7AHwAEAmABgFgAoUgRmuoDcBDKAAmbYF425aBuakVoBOABQASAEQBvMTgCWeDAEpmAXykr+QA==>